app/lib/RouteResolver.js

```diff
@@ -85,9 +81,6 @@ RouteResolver.prototype.register = function register(method, path, response) {
 
   this.routes.push(route);
 
-  // it is necessary to mount catch all route here to avoid overriding middleware
-  listen.call(this);
-
```

Comment:
This was a bad smell, glad to see you were able to get the resolver listening upon instantiation.